engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Array.java

```diff
@@ -129,21 +121,6 @@ boolean isArrayElementReadable(long index) {
     return index < getArraySize() && index >= 0;
   }
 
-  @ExportMessage
-  void writeArrayElement(long index, Object value) {
-    items[(int) index] = value;
-  }
-
-  @ExportMessage
-  boolean isArrayElementModifiable(long index) {
-    return isArrayElementReadable(index);
-  }
-
-  @ExportMessage
-  boolean isArrayElementInsertable(long index) {
-    return false;
-  }
```

Comment:
I decided to remove these, since our arrays are meant to be immutable by design.

Curiously, still, `Array.copy` is able to overwrite contents of our arrays even if they report being 'unmodifiable'. Very weird. @JaroslavTulach do you think this may be a bug? I assume it should be somehow possible to create a completely read-only array which does not support writing (possibly can be done by wrapping it in an `UnmodifiableList`).
OTOH we are actually currently relying on this, as we use `Array.copy` for implementing many of our Vector operations.